File: LeetCode/Medium/2410-Maximum_Matching_Of_Players_With_Trainers/2410-MaximumMatchingOfPlayersWithTrainers.cpp
#include <algorithm>
#include <vector>

using namespace std;

class Solution
{
public:
    int matchPlayersAndTrainers(vector<int> &players, vector<int> &trainers)
    {
        int n = players.size();
        int m = trainers.size();

        sort(players.begin(), players.end());
        sort(trainers.begin(), trainers.end());

        int j = 0;
        int count = 0;

        for (int i = 0; i < n; i++)
        {
            while (j < m && players[i] > trainers[j])
                j++;

            if (j < m)
            {
                count++;
                j++;
            }
            else
                break;
        }

        return count;
    }
};

File: LeetCode/Medium/2410-Maximum_Matching_Of_Players_With_Trainers/2410-MaximumMatchingOfPlayersWithTrainers.py
class Solution:
    def matchPlayersAndTrainers(self, players: list[int], trainers: list[int]) -> int:
        n = len(players)
        m = len(trainers)
        players.sort()
        trainers.sort()

        j = 0
        count = 0

        for i in range(n):
            while j < m and players[i] > trainers[j]:
                j += 1

            if j < m:
                count += 1
                j += 1
            else:
                break

        return count


a = Solution()
print(a.matchPlayersAndTrainers(players=[4, 7, 9], trainers=[8, 2, 5, 8]))
print(a.matchPlayersAndTrainers(players=[1, 1, 1], trainers=[10]))
